### PR TITLE
Rewrite project structure so that goquery can be imported as a library

### LINKED ISSUE #110: Rewrite project structure so that goquery can be imported as a library
As of @directionless's suggestion, it makes more sense for goquery to be imported as a library, rather than needing to clone and modify the source to get any integration working.

`main.go` should be reworked so that the entry point is in goquery.go with a call to Start(api GoQueryAPI, config Config) where api is either a custom or referenced built in API and config is a populated Config struct.

An example wrapper `example.go` can be used to demonstrate integration, and also used to `go run` to see an example working quickly.

### PR DESCRIPTION
**Pull request by AbGuthrie**
Instead of running goquery from the conventional cmd/main.go, a custom main file will now need to be created as goquery should behave as a library important. This makes more sense from the viewpoint of custom API integrations and the open source nature of the project.

This commit contains the following changes:

- Removes cmd/main.go in with just oquery.go (now importable to run Run function)
- Removed previous global api module. Now the api folder is just the available integrations, and an API struct implement GoQueryAPI is now passed through the program from the Run function
- Renamed api/mock and osctrl integrations to client.go
- Changed command structure to recieve an api struct and the configuration state - no longer relying on global modules
- Moved schedule query and wait to util, as there is no api module anymore. The utils function is just handed an api struct and knows how to handle ctrl c interupt
- Config no longer parses json files as the options must be configured by the function importing goquery (users will need to make their own option config parsers in their own main files)
- Created models folder with Rows and the GoQueryAPI interface definitions
- goquery must now be called with a custom go file which can be copy pasted from the readme example

Closes #110

### CODE REVIEW

File: commands/change_directory.go [hunk 1]
```diff
@@ -44,19 +46,17 @@ func changeDirectory(cmdline string) error {
...
 	}
 
 	if len(results) == 1 {
 		return hosts.SetCurrentHostDirectory(requestedDirectory)
-	} else {
-		return fmt.Errorf("No such directory")
 	}
 
-	return nil
+	return fmt.Errorf("No such directory")
```
Comment (obelisk) [inline commands/change_directory.go]: I would generally say this if logic should be reverse (early exit on error not success)

File: config/config.go [hunk 2]
```diff
@@ -21,11 +16,11 @@ type Alias struct {
...
 type Config struct {
-	DebugEnabled     bool             `json:"debugEnabled"`
-	APIDriver        string           `json:"apiDriver"`
-	Experimental     bool             `json:"experimental"`
-	CurrentPrintMode PrintMode        `json:"printMode"`
-	Aliases          map[string]Alias `json:"aliases"`
+	DebugEnabled bool             `json:"debugEnabled"`
+	APIDriver    string           `json:"apiDriver"`
+	Experimental bool             `json:"experimental"`
+	PrintMode    PrintMode        `json:"printMode"`
```
Comment (obelisk) [inline config/config.go]: Is it fine to overload the name and type here?

File: models/rows.go [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package models
+
+// Rows is the Osquery response shape parsed from API, or built manually to utilize printers
+type Rows = []map[string]string
```
Comment (obelisk) [inline models/rows.go]: I could cry tears of joy

File: utils/printer.go [hunk 4]
```diff
@@ -6,11 +6,11 @@ import (
...
 	"sort"
 	"strings"
-)
 
-type Rows = []map[string]string
+	"github.com/AbGuthrie/goquery/models"
+)
 
-func prettyPrintQueryResultsJSON(results Rows) {
+func prettyPrintQueryResultsJSON(results models.Rows) {
```
Comment (obelisk) [inline utils/printer.go]: Can we just import as Rows as the top so we don't need to specify each time?

File: utils/scheduleQueryAndWait.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
```
Comment (obelisk) [inline utils/scheduleQueryAndWait.go]: We (I) broke this before last time I refactored it. We definitely need to check that it still works.

File: utils/utils.go [hunk 6]
```diff
@@ -7,9 +7,8 @@ import (
 
 // PrettyPrintQueryResults prints a given []result map set to standard out
 // taking into consideration the current state.go's print mode
-func PrettyPrintQueryResults(results []map[string]string) {
-	currentConfig := config.GetConfig()
-	switch currentConfig.CurrentPrintMode {
+func PrettyPrintQueryResults(results []map[string]string, printMode config.PrintMode) {
```
Comment (obelisk) [inline utils/utils.go]: This should use the Rows type (cries tears of joy)

**Comment by obelisk** [comment]
@directionless

File: README.md [hunk 7]
```diff
@@ -73,6 +73,65 @@ List the files in the current directory. The current directory is set by using t
...
+	"github.com/AbGuthrie/goquery/config"
+	"github.com/AbGuthrie/goquery/hosts"
+	"github.com/AbGuthrie/goquery/models"
+)
+
+func main() {
+	// 1. Provide something that implements the required models/GoQueryAPI interface,
+	//	  or use a supported built in (see `api/mock` for example implementation)
+	api := myCustomAPI{}
+	// osctrlAPI, err := osctrl.CreateOSctrlAPI(true)	// import goquery/api/mock
```
Comment (obelisk) [inline README.md]: These are backwards

**Comment by AbGuthrie** [comment]
Good to go, @obelisk  PTAL

**Comment by obelisk** [comment]
Can we reintroduce alias loading code before we merge? That's such a valuable feature. We also need to fix the make file so we can easily make a mock binary

**Pull request merged**